src/awkward/cpp-headers/awkward/GrowableBuffer.h

```diff
@@ -142,6 +151,23 @@ namespace awkward {
       }
     }
 
+    /// @brief 'copy_as' specialization of a 'std::complex' template type.
+    /// Fills (one panel) GrowableBuffer<std::complex> with the
+    /// elements of (possibly multi-panels) GrowableBuffer<PRIMITIVE>.
+    ///
+    /// Changes the data type from `PRIMITIVE` to `std::complex`/
+    template <typename TO_PRIMITIVE>
+    typename std::enable_if<awkward::is_tt<std::complex, TO_PRIMITIVE>::value>::type
+    copy_as(TO_PRIMITIVE* to_ptr, size_t offset) {
+      for (size_t i = 0; i < length_; i++) {
+        double val = static_cast<double>(ptr_.get()[i]);
+        to_ptr[offset++] = TO_PRIMITIVE(val);
+      }
+      if (next_) {
+        next_->copy_as(to_ptr, offset);
+      }
+    }
+
```

Comment:
@jpivarski - this template specialisation makes the Linux compiler happy. However, now I realise that the `copy_as` allows pretty much any type conversion including those that would lead to a data loss... We leave it to a user to make a sensible decision, but I'd rather make it more restrictive implementation. What do you think?

Reply:
Is "the user" in this case code that uses GrowableBuffer? If so, that's just us: LayoutBuilder is user-facing (for users that are downstream developers of C++ applications, or maybe users of RDataFrame), but GrowableBuffer is not. Although the header-only files expose GrowableBuffer as much as they do LayoutBuilder, we haven't documented it or encouraged its use.

If you're asking whether we can have an internal function that converts type X to type Y, possibly losing a lot of data (e.g. X = uint64 and Y = int8, or X = double), then that's fine. A converter function is going to convert the types you give it. Especially since GrowableBuffer is internal, we can do this.

Reply:
ok, thanks!